test/unit/mdc-menu/menu.foundation.test.js

```diff
@@ -80,1165 +51,431 @@ test('exports cssClasses', () => {
   const {foundation, mockAdapter} = setupTest();
-  const handlers = captureHandlers(mockAdapter, 'registerInteractionHandler');
   const clock = lolex.install();
-  const target = {};
-  const preventDefault = td.func('event.preventDefault');
-  td.when(mockAdapter.getIndexForEventTarget(target)).thenReturn(0, 1);
-
-  foundation.init();
-  handlers.keydown({target, key: 'Space', preventDefault});
-  handlers.keyup({target, key: 'Space'});
-  handlers.keydown({target, key: 'Space', preventDefault});
-  handlers.keyup({target, key: 'Space'});
-  clock.tick(numbers.SELECTED_TRIGGER_DELAY);
-  td.verify(mockAdapter.notifySelected({index: 0}), {times: 1});
+  const key = {key: 'Space', target: {tagName: 'li'}, preventDefault: td.func('preventDefault')};
+  td.when(mockAdapter.elementContainsClass(key.target, listClasses.LIST_ITEM_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getElementIndex(key.target)).thenReturn(0);
+  td.when(mockAdapter.getParentElement(key.target)).thenReturn(key.target);
+  td.when(mockAdapter.elementContainsClass(key.target, cssClasses.MENU_SELECTION_GROUP)).thenReturn(true);
+  td.when(mockAdapter.getSelectedElementIndex(key.target)).thenReturn(0);
+
+  foundation.handleKeydown(key);
+  clock.tick(numbers.TRANSITION_CLOSE_DURATION);
 
+  td.verify(mockAdapter.removeClassFromElementAtIndex(0, cssClasses.MENU_SELECTED_LIST_ITEM), {times: 1});
+  td.verify(mockAdapter.addClassToElementAtIndex(0, cssClasses.MENU_SELECTED_LIST_ITEM), {times: 1});
   clock.uninstall();
 });
 
-test('on spacebar keyup does works if DOM3 keyboard events are not supported', () => {
+test('handleKeydown space/enter key inside of a selection group with no element selected', () => {
   const {foundation, mockAdapter} = setupTest();
-  const handlers = captureHandlers(mockAdapter, 'registerInteractionHandler');
   const clock = lolex.install();
-  const target = {};
-  const preventDefault = td.func('event.preventDefault');
-  td.when(mockAdapter.getIndexForEventTarget(target)).thenReturn(0);
+  const key = {key: 'Space', target: {tagName: 'li'}, preventDefault: td.func('preventDefault')};
+  td.when(mockAdapter.elementContainsClass(key.target, listClasses.LIST_ITEM_CLASS)).thenReturn(true);
+  td.when(mockAdapter.getElementIndex(key.target)).thenReturn(0);
+  td.when(mockAdapter.getParentElement(key.target)).thenReturn(key.target);
+  td.when(mockAdapter.elementContainsClass(key.target, cssClasses.MENU_SELECTION_GROUP)).thenReturn(true);
+  td.when(mockAdapter.getSelectedElementIndex(key.target)).thenReturn(-1);
 
-  foundation.init();
-  handlers.keydown({target, keyCode: 32, preventDefault});
-  handlers.keyup({target, keyCode: 32});
-  clock.tick(numbers.SELECTED_TRIGGER_DELAY);
-  td.verify(mockAdapter.notifySelected({index: 0}));
+  foundation.handleKeydown(key);
+  clock.tick(numbers.TRANSITION_CLOSE_DURATION);
 
+  td.verify(mockAdapter.removeClassFromElementAtIndex(0, cssClasses.MENU_SELECTED_LIST_ITEM), {times: 0});
```

Comment:
Replace `0` first argument to adapter call with `td.matchers.isA(Number)`?

(Same for the next 2 tests in `{times: 0}` checks, to ensure they're not calling it at all with any value)